contracts/liquidity_hub/fee_collector/src/queries.rs

```diff
@@ -12,3 +17,84 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
     let config = CONFIG.load(deps.storage)?;
     Ok(config)
 }
+
+pub fn query_accrued_fees(deps: Deps, collect_fees_for: CollectFeesFor) -> StdResult<Vec<Asset>> {
+    let mut query_fees_messages: Vec<Asset> = Vec::new();
+
+    match collect_fees_for {
+        CollectFeesFor::Contracts { contracts } => {
+            for contract in contracts {
+                let response = query_fees_for_vault(&deps, contract)?;
+
+                query_fees_messages.push(response.fees);
+            }
+        }
+        CollectFeesFor::Factory {
+            factory_addr,
+            factory_type,
+        } => {
+            let factory = deps.api.addr_validate(factory_addr.as_str())?;
+            let mut assets = query_fees_for_factory(&deps, &factory, factory_type)?;
+
+            query_fees_messages.append(&mut assets);
+        }
+    }
+
+    Ok(query_fees_messages)
+}
+
+fn query_fees_for_vault(deps: &Deps, vault: String) -> StdResult<ProtocolVaultFeesResponse> {
+    deps.querier
+        .query::<ProtocolVaultFeesResponse>(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: vault,
+            msg: to_binary(&vault_network::vault::QueryMsg::ProtocolFees { all_time: false })?,
+        }))
+}
+
+fn query_fees_for_pair(deps: &Deps, pair: String) -> StdResult<ProtocolPairFeesResponse> {
+    deps.querier
+        .query::<ProtocolPairFeesResponse>(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: pair,
+            msg: to_binary(&vault_network::vault::QueryMsg::ProtocolFees { all_time: false })?,
```

Comment:
The msg here is incorrect, should be `terraswap::pair::QueryMsg::ProtocolFees` instead

Reply:
:heavy_check_mark: